From d3814d47ec92091dbf524bdabaa3c1fc71ed2c80 Mon Sep 17 00:00:00 2001
From: lazedo <luis.azedo@factorlusitano.com>
Date: Tue, 24 Jan 2017 02:58:54 +0000
Subject: [PATCH] presence: free pointer after usage

discovered when using tlsf memory manager.
(cherry picked from commit dc518f6973f81b71264036b138dd6293eef58ed2)
---
 modules/presence/notify.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/modules/presence/notify.c b/modules/presence/notify.c
index 04ee88f12..9e5037517 100644
--- a/modules/presence/notify.c
+++ b/modules/presence/notify.c
@@ -982,9 +982,8 @@ dlg_t* ps_build_dlg_t(subs_t* subs)
 			pkg_free(tmp);
 			goto error;
 		}
+		td->send_sock = grep_sock_info (&host, (unsigned short) port, (unsigned short) proto);
 		pkg_free(tmp);
-		td->send_sock = grep_sock_info (
-			&host, (unsigned short) port, (unsigned short) proto);
 	}
 	
 	return td;
-- 
2.11.0

